packages/viewer/src/store/modules/app/index.ts

```diff
@@ -51,15 +51,26 @@ const initiateUrlParsing: AppState = {
 
 const parseLegacyUrlParams: AppState = {
     name: AppStateNames.LegacyParsing,
-    isFulfilled: () => !isLegacyParams(window?.location?.search),
+    isFulfilled: () => {
+        // Since it's not a ref here, it only checks it once and never check if it has changed
+        // we should have a small ref which contains windows.location
+        return !isLegacyParams(window?.location?.search)
+    },
     next: () => {
         return initiateUrlParsing
     },
 }
 
 const configLoaded: AppState = {
     name: AppStateNames.ConfigLoaded,
-    isFulfilled: () => true, // there's always a topic set, so no need to check if topicStore.current is defined
+    isFulfilled: () => {
+        // When initializing the config, we could run into issues where the URL parser was active while
+        // the app was setting the default background, writing the topic default background layer over the URL
+        // parameter. This check ensure the topics have set what they need to before we activate the URL sync tool
```

Comment:
The URL sync is not enabled at this stage of the app, so that should never happen.
Did you encounter this bug?